shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -30,56 +32,117 @@ function _getLocalStorageObject(): Storage {
  * @returns {Storage} Returns storage object verified that it is usable
  */
 function _getVerifiedStorageObject(storageType: StorageType): Storage {
+    let result = null;
+    const storageTypeName = storageType === StorageType.LocalStorage ? "localStorage" : "sessionStorage";
+    
     try {
-        if (isNullOrUndefined(getGlobal())) {
-            return null;
+        // Default to false - assume storage is not available
+        let canAccessStorage = false;
+        const gbl = getGlobal();
+        
+        // Only proceed if we have a global object
+        if (!isNullOrUndefined(gbl)) {
+            // Try the safest method first (property descriptor)
+            try {
+                const descriptor = objGetOwnPropertyDescriptor(gbl, storageTypeName);
+                if (descriptor && descriptor.get) {
+                    canAccessStorage = true;
+                }
+            } catch (e) {
+                // If descriptor check fails, try direct access
+                // This will be caught by the outer try-catch if it fails
+                canAccessStorage = !!gbl[storageTypeName];
+            }
         }
-        let uid = (new Date).toString();
-        let storage: Storage = getGlobalInst(storageType === StorageType.LocalStorage ? "localStorage" : "sessionStorage");
-        let name:string = _storagePrefix + uid;
-        storage.setItem(name, uid);
-        let fail = storage.getItem(name) !== uid;
-        storage.removeItem(name);
-        if (!fail) {
-            return storage;
+        
+        // If we determined storage might be accessible, verify it works
+        if (canAccessStorage) {
+            try {
+                const uid = (new Date).toString();
+                const storage: Storage = getGlobalInst(storageTypeName);
+                if (!storage) {
+                    return null;
+                }
+                const name = _storagePrefix + uid;
+                
+                storage.setItem(name, uid);
+                const success = storage.getItem(name) === uid;
+                storage.removeItem(name);
+                
+                if (success) {
+                    // Create a wrapped storage object that protects write operations
+                    const originalStorage = storage;
+                    
+                    // Helper to create storage operation methods with consistent error handling
+                    const _createStorageOperation = function<T>(operationName: string, resetOnError: boolean, defaultValue?: T): Function {
+                        return function(...args: any[]): T {
+                            // Let exceptions propagate to the caller
+                            return originalStorage[operationName].apply(originalStorage, args);
```

Comment:
Add try / catch and include the error handling like previous but then rethrow the exception to the caller